experiment/compatibility-versions/e2e-k8s-compatibility-versions.sh

```diff
@@ -391,7 +391,12 @@ main() {
   UPGRADE_SCRIPT="${UPGRADE_SCRIPT:-${PWD}/../test-infra/experiment/compatibility-versions/kind-upgrade.sh}"
   echo "Upgrading cluster with ${UPGRADE_SCRIPT}"
 
-  upgrade_cluster_components
+  upgrade_cluster_components || res=$?
+  # If upgrade fails, we shouldn't run tests but should still clean up.
```

Comment:
Not sure if we should just continue or exit here, either way should work imo, but since upgrade fails not sure if it's valid to run the tests.